### ISSUE TITLE
[RFE] Shared Credentials for F5 Hosts

### ISSUE THREAD

**Issue by jeremy-church1** [f5devcentral/vscode-f5#180]
I have several BigIP controllers I connect to regularly. I use the same credentials on all of them which changes daily. It would be extremely convenient if I could enter those credentials once in one place instead of on each device.

Here's a more detailed explanation of the user interface:
- Add a section for "Credentials" where credentials can be added/deleted/changed.
  - Each credential would have an ID (name or label), a username and password.
  - Context menu for each credential entry would have options to clear password, set password and delete.
- Within an F5 HOSTS entry, add a "credentials" field which refers back to the ID of a specific credential entry.
- When connecting to a device, the credentials field takes priority when retrieving the username and password.

Thanks

**Comment by DumpySquare** [comment]
I'm gonna have to really think about how to handle this request.  

In the mean time.  You should be able to globally set the password by re-importing all the configured devices.  A template and a simple global replace should make this happen in seconds.  

Check this out:
https://f5devcentral.github.io/vscode-f5/#/device_importing?id=json-object-import-structure

**Comment by jeremy-church1** [comment]
> In the mean time. You should be able to globally set the password by re-importing all the configured devices. A template and a simple global replace should make this happen in seconds.
> 
> Check this out: https://f5devcentral.github.io/vscode-f5/#/device_importing?id=json-object-import-structure

When trying to work around my issue of clearing passwords, I came across the JSON structure and tried it. It didn't work for me. I mentioned it very briefly in the clear password issue I created yesterday. Honestly if clear passwords worked for me or the JSON import worked, I may not have even thought of this feature. I was just thinking how much of a pain it would be if I ever wanted to connect to more of our devices of which we have many.

Thanks for replying. Even with the problems, the extension is really good and extremely helpful.
